routes/ecom/modules/list-payments.js

```diff
@@ -3,12 +3,40 @@
 // generate default payment gateway object
 const newPaymentGateway = require(process.cwd() + '/lib/new-payment-gateway')
 
+const checkDiscount = (paymentGateway, items) => {
+  const discount = paymentGateway.discount
+  const productIds = discount?.product_ids
+  if (items && items.length) {
+    if (discount.apply_at !== 'freight' && discount.value > 0 && productIds && productIds.length) {
+      let discountValue = 0
+      items.forEach(item => {
+        if (productIds.includes(item.product_id)) {
+          discountValue += (item.quantity) * (item.final_price || item.price)
+        }
+      })
+      if (discountValue && discount.type === 'percentage') {
```

Comment:
Se for desconto em valor fixo usa `Math.min`

Reply:
Quer dizer, sendo valor fixo o desconto aplicado será até 200 reais por exemplo, mas limitado ao valor dos itens selecionados

Reply:
> Na verdade isso é improvável que seja útil também, se preferir pode não tratar esse cenário mas nesse caso joga esse `discount.type === 'percentage'` pra cima, não precisaria chegar até aqui, somar o `discountValue`...